packages/calypso-e2e/src/lib/pages/plugins-page.ts

```diff
@@ -0,0 +1,34 @@
+import { Page } from 'playwright';
+import { getCalypsoURL } from '../../data-helper';
+
+const selectors = {
+	sectionTitle: ( section: string ) => `.plugins-browser-list__title:text("${ section }")`,
+};
+
+/**
+ * Plugins page https://wordpress.com/plugins/.
+ */
+export class PluginsPage {
+	private page: Page;
+
+	/**
+	 * Constructs an instance.
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+	}
+
+	/**
+	 * Visit /plugins/*
+	 */
+	async visit( route?: string ): Promise< void > {
+		await this.page.goto( getCalypsoURL( `${ route || '' }` ) );
+	}
```

Comment:
With `visit` methods, it's common convention just couple the content of the HTTP route to the method!
We also tend to rely on the account pre-authentication and most of our test accounts just having one site to handle the redirect to the correct site.

All of that prevents the test scripts and writers from having to try to construct the right URL in the testing spec!

So, in this case, you could define `visit` as...

```suggestion
	async visit(): Promise< void > {
		await this.page.goto( getCalypsoURL( 'plugins' ) );
	}
```

Reply:
Thanks, I was struggling with how to manage the fact we have a few different routes to visit that all load the same component in a different context.

```
/plugins
/plugins/:site
/plugins/:plugin
/plugins/:plugin/:site
/plugins/:category
/plugins/:category/:site
```

I almost put `/plugins/${route}` but hiding the `/plugins` part made tests harder to read. A compromise might be to have `${route || 'plugins' }` Although I'm now leaning towards having separate visit functions for each route leaving the standard visit method to stick with convention. I'll do this in a follow up, got a few coming this week.

> We also tend to rely on the account pre-authentication and most of our test accounts just having one site to handle the redirect to the correct site.

Curious how this works - does this require the page to auto redirect if the account only has one site? Does the plugins page do this? (I'm not aware either way)

Reply:
>Curious how this works - does this require the page to auto redirect if the account only has one site? Does the plugins page do this? (I'm not aware either way)

Hmm, yeah, I think it has kind of worked off that assumption in the past, but as I'm thinking about this, this is kind of a bad assumption. 🤔  Not all pages have that logic, and it looks like `plugins` actually doesn't. Bummer!

I think your approach for mentioning the site explicitly is a good approach for now then, sorry for the confusion!

>I was struggling with how to manage the fact we have a few different routes to visit that all load the same component in a different context

Oof, I didn't recognize this! This certainly complicates things a bit, ha. I think because of the uniqueness of this page (in it's many variations of the same component), convention can be bucked and whatever you think makes the tests read the cleanest is great!

Our `visit` design kind of worked off the assumption that there was a fixed 1:1 route to component under test, but that assumption also is clearly shaky. 

I wrote a task up for us to re-look at the old `visit` convention (#61370). It isn't something I think we gave a ton of attention to, and it warrants a touch up. 😄